interconnect/src/main/java/fe/linksheet/interconnect/BaseParceledListSlice.java

```diff
@@ -0,0 +1,196 @@
+package fe.linksheet.interconnect;
+
+import android.os.Binder;
+import android.os.Build;
+import android.os.IBinder;
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.os.RemoteException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Transfer a large list of Parcelable objects across an IPC.  Splits into
+ * multiple transactions if needed.
+ * Caveat: for efficiency and security, all elements must be the same concrete type.
+ * In order to avoid writing the class name of each object, we must ensure that
+ * each object is the same type, or else unparceling then reparceling the data may yield
+ * a different result if the class name encoded in the Parcelable is a Base type.
+ * See b/17671747.
+ */
+abstract class BaseParceledListSlice<T> implements Parcelable {
```

Comment:
Copied and slightly modified from AOSP, since even the hidden API JAR doesn't seem to let AIDLs use hidden APIs.